tests/test_bukuDb.py

```diff
@@ -488,6 +489,125 @@ def test_compactdb(setup):
     assert bdb.get_rec_by_id(2) == (2, 'https://test.com:8080', 'test', ',es,est,tes,test,', 'a case for replace_tag test', 0)
     assert bdb.get_rec_by_id(3) is None
 
+
+@pytest.mark.parametrize(
+    'index, low, high, is_range, delay_commit, empty_database, input_retval',
+    product(
+        [-1000000, -1, 0, 1, 3, 4, 1000000],
```

Comment:
As I mentioned,
- if is_range is True, we care only about low and high (index not even checked)
- if is_range is False, we care only about index (low/high not even checked)